swatch-metrics/src/main/java/com/redhat/swatch/metrics/service/PrometheusMeteringController.java

```diff
@@ -239,6 +239,35 @@ private void createEventFromDataAndSend(
       role = null;
     }
 
+    boolean is3rdPartyMigrated = Boolean.parseBoolean(labels.get("conversions_success"));
+
+    var matchingTags =
+        SubscriptionDefinition.getAllProductTagsWithPaygEligibleByRoleOrEngIds(
+            role, productIds, null, is3rdPartyMigrated);
+
+    Optional<String> firstTag = matchingTags.stream().findFirst();
```

Comment:
As an alternative to finding first, let's iterate over this since, in the future, if it returns two producttags and the second producttag matches the product tag in context, it will skip without even checking.

Reply:
Updated to remove `findFirst()` usage.  Instead, I check the size of the Set<String> matchingTags.  If the size is anything besides 1, log a warning and don't continue event creation.  A size 0 means that there's no match, and size > 1 means we couldn't determine a unique product tag - we don't want to guess which one to use, and we don't want to send it twice.